discovery-provider/src/monitors/monitors.py

```diff
@@ -0,0 +1,177 @@
+    'func': get_database_liveness,
+    'type': 'bool'
+}
+DATABASE_SIZE = {
+    'name': 'database_size',
+    'func': get_database_size,
+    'ttl': 60 * 2,
+    'type': 'int'
+}
+DATABASE_CONNECTIONS = {
+    'name': 'database_connections',
+    'func': get_database_connections,
+    'type': 'int'
+}
+DATABASE_CONNECTION_INFO = {
+    'name': 'database_connection_info',
+    'func': get_database_connection_info,
+    'type': 'json'
+}
+
+TOTAL_MEMORY = {
+    'name': 'total_memory',
+    'func': get_total_memory,
+    'ttl': 60 * 2,
+    'type': 'int'
+
+}
+USED_MEMORY = {
+    'name': 'used_memory',
+    'func': get_used_memory,
+    'ttl': 60 * 2,
+    'type': 'int'
+}
+
+FILESYSTEM_SIZE = {
+    'name': 'filesystem_size',
+    'func': get_filesystem_size,
+    'ttl': 60 * 5,
+    'type': 'int'
+
+}
+FILESYSTEM_USED = {
+    'name': 'filesystem_used',
+    'func': get_filesystem_used,
+    'ttl': 60 * 5,
+    'type': 'int'
+}
+
+RECEIVED_BYTES_PER_SEC = {
+    'name': 'received_bytes_per_sec',
+    'func': get_received_bytes_per_sec,
+    'type': 'float'
+}
+
+TRANSFERRED_BYTES_PER_SEC = {
+    'name': 'transferred_bytes_per_sec',
+    'func': get_transferred_bytes_per_sec,
+    'type': 'float'
+}
+
+REDIS_NUM_KEYS = {
+    'name': 'redis_num_keys',
+    'func': get_redis_num_keys,
+    'ttl': 60 * 5,
+    'type': 'int'
+}
+REDIS_USED_MEMORY = {
+    'name': 'redis_used_memory',
+    'func': get_redis_used_memory,
+    'ttl': 60 * 5,
+    'type': 'int'
+}
+REDIS_TOTAL_MEMORY = {
+    'name': 'redis_total_memory',
+    'func': get_redis_total_memory,
+    'ttl': 60 * 5,
+    'type': 'int'
+}
+
+MONITORS = {
+    'DATABASE_LIVENESS': DATABASE_LIVENESS,
+    'DATABASE_SIZE': DATABASE_SIZE,
+    'DATABASE_CONNECTIONS': DATABASE_CONNECTIONS,
+    'DATABASE_CONNECTION_INFO': DATABASE_CONNECTION_INFO,
+    'TOTAL_MEMORY': TOTAL_MEMORY,
+    'USED_MEMORY': USED_MEMORY,
+    'FILESYSTEM_SIZE': FILESYSTEM_SIZE,
+    'FILESYSTEM_USED': FILESYSTEM_USED,
+    'RECEIVED_BYTES_PER_SEC': RECEIVED_BYTES_PER_SEC,
+    'TRANSFERRED_BYTES_PER_SEC': TRANSFERRED_BYTES_PER_SEC,
+    'REDIS_NUM_KEYS': REDIS_NUM_KEYS,
+    'REDIS_USED_MEMORY': REDIS_USED_MEMORY,
+    'REDIS_TOTAL_MEMORY': REDIS_TOTAL_MEMORY
+}
+
+def get_monitor_redis_key(monitor):
+    return f"{MONITORING_REDIS_PREFIX}:{monitor['name']}"
+
+def parse_value(monitor, value):
+    """
+    Parses a string value into the corresponding type
+
+    Args:
+        monitor: dict The monitor dictionary qwith name, func, ttl, and type
+        value: string The value to parse
+    """
+    try:
+        if monitor['type'] == 'bool':
+            return value == 'True'
+        elif monitor['type'] == 'int':
+            return int(value)
+        elif monitor['type'] == 'float':
+            return float(value)
+        elif monitor['type'] == 'json':
+            return json.loads(value)
+        else: # string
+            return str(value)
+    except Exception as e:
+        return str(value)
```

Comment:
if an exception occurs and we return str(value), would that cause a monitoring failure later on in the flow? should we handle exception cases differently?

Reply:
the parser here is used when reading values from the monitor and i thought it better to just return the string dumped from redis rather than crashing. wdyt?

Reply:
yup yup, good call